src/Variable.cpp

```diff
@@ -200,6 +200,26 @@ VariablePtr Variable::subvariable(Space space, std::string_view baseName, Space
   }
 }
 
+VariablePtr Variable::subvariable(Space space, Space shift) const
+{
+  if(!(space * shift <= space_))
+    throw std::runtime_error("[Variable::subvariable] Invalid space and shift dimension");
```

Comment:
I think we should build more helpful exception messages. This one should display: `space` dimensions, `shift` dimensions and `space_` dimensions.

Reply:
Done in a702cada5efdf4ee0f0bb6e7b5d0e9dc6f81648b